client/java/src/main/java/com/linecorp/centraldogma/internal/client/AbstractWatcher.java

```diff
@@ -296,18 +313,29 @@ private void notifyListeners() {
         }
 
         final Latest<T> latest = this.latest;
+        final List<CompletableFuture<Boolean>> futures = new ArrayList<>();
         for (Map.Entry<BiConsumer<? super Revision, ? super T>, Executor> entry : updateListeners) {
             final BiConsumer<? super Revision, ? super T> listener = entry.getKey();
             final Executor executor = entry.getValue();
-            executor.execute(() -> {
+            final CompletableFuture<Boolean> future = CompletableFuture.supplyAsync(() -> {
                 try {
                     listener.accept(latest.revision(), latest.value());
+                    return true;
                 } catch (Exception e) {
                     logger.warn("Exception thrown for watcher ({}/{}{}): rev={}",
                                 projectName, repositoryName, pathPattern, latest.revision(), e);
+                    return false;
                 }
-            });
+            }, executor);
+            futures.add(future);
         }
+
+        CompletableFutures.allAsList(futures).thenAccept(results -> {
+            final boolean result = results.stream().allMatch(x -> x);
```

Comment:
Shouldn't we update the `latestNotifiedRevision` regardless of a listener throws an exception?

Reply:
I've added another metric which updates the revision irregardless of whether the notify succeeds.

I think the two metrics combined can give users a good idea of whether properties have been propagated correctly.